pkg/loader/model.go

```diff
@@ -167,94 +177,69 @@ func (dml *DML) oldPrimaryKeyValues() []interface{} {
 
 // TableName returns the fully qualified name of the DML's table
 func (dml *DML) TableName() string {
+	if dml.DestDBType == OracleDB {
+		return fmt.Sprintf("%s.%s", dml.Database, dml.Table)
+	}
 	return quoteSchema(dml.Database, dml.Table)
 }
 
-// OracleTableName returns the fully qualified name of the DML's table in oracle db
-func (dml *DML) OracleTableName() string {
-	return fmt.Sprintf("%s.%s", dml.Database, dml.Table)
-}
-
 func (dml *DML) updateSQL() (sql string, args []interface{}) {
 	builder := new(strings.Builder)
 
 	fmt.Fprintf(builder, "UPDATE %s SET ", dml.TableName())
-
+	oracleHolderPos := 1
 	for _, name := range dml.columnNames() {
 		if len(args) > 0 {
 			builder.WriteByte(',')
 		}
 		arg := dml.Values[name]
-		fmt.Fprintf(builder, "%s = ?", quoteName(name))
+		if dml.DestDBType == OracleDB {
+			fmt.Fprintf(builder, "%s = :%d", escapeName(name), oracleHolderPos)
+			oracleHolderPos++
+		} else {
+			fmt.Fprintf(builder, "%s = ?", quoteName(name))
+		}
 		args = append(args, arg)
 	}
 
 	builder.WriteString(" WHERE ")
 
-	whereArgs := dml.buildWhere(builder)
+	whereArgs := dml.buildWhere(builder, oracleHolderPos)
 	args = append(args, whereArgs...)
-
-	builder.WriteString(" LIMIT 1")
-	sql = builder.String()
-	return
-}
-
-func (dml *DML) oracleUpdateSQL() (sql string) {
-	builder := new(strings.Builder)
-
-	fmt.Fprintf(builder, "UPDATE %s SET ", dml.OracleTableName())
-
-	for i, name := range dml.columnNames() {
-		if i > 0 {
-			builder.WriteByte(',')
-		}
-		value := dml.Values[name]
-		if value == nil {
-			fmt.Fprintf(builder, "%s = NULL", escapeName(name))
-		} else {
-			fmt.Fprintf(builder, "%s = %s", escapeName(name), genOracleValue(dml.UpColumnsInfoMap[name], value))
-		}
+	if dml.DestDBType == OracleDB {
+		builder.WriteString(" AND rownum <=1")
+	} else {
+		builder.WriteString(" LIMIT 1")
 	}
-
-	builder.WriteString(" WHERE ")
-
-	dml.buildOracleWhere(builder)
-	builder.WriteString(" AND rownum <=1")
-
 	sql = builder.String()
 	return
 }
 
-func (dml *DML) buildWhere(builder *strings.Builder) (args []interface{}) {
+func (dml *DML) buildWhere(builder *strings.Builder, oracleHolderPos int) (args []interface{}) {
```

Comment:
However, I'm a bit afraid this combination will make drainer has a lower effiency. We will do a lot of extra `if check` however half of them are useless because we won't change destDBType when we runs drainer.
refer: https://zhuanlan.zhihu.com/p/143275246 提前进行分支预测